### PR TITLE
JBPM-9151 API return 200 if containerId does not exist

### PR DESCRIPTION
**Pull request by fjtirado**
Check if deployment id is available in the in memory list and throw
proper exception if not.

### CODE REVIEW

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/BPMN2DataServicesTest.java [hunk 1]
```diff
@@ -347,6 +347,7 @@ public void testHumanTaskProcessBeforeAndAfterUndeploy() {
 
         // now let's undeploy the unit
         deploymentService.undeploy(deploymentUnit);
+        units.remove(deploymentUnit);
```
Comment (elguardian) [inline jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/BPMN2DataServicesTest.java:350]: why this ? the change to RuntimeDataServiceImpl.java looks to me like completely compatible regarding this.

**Comment by elguardian** [comment]
some minor question.

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/BPMN2DataServicesTest.java (hunk 1 above)
Comment (fjtirado) [reply jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/BPMN2DataServicesTest.java:350]: This change was suggested by Gonzalo. The target is to keep client units array consistent with the server.

Comment (gmunozfe) [reply jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/BPMN2DataServicesTest.java:350]: Yes, that's the reason, it's the same in other tests:
https://github.com/kiegroup/jbpm/blob/74b525d1454e80de85ae02a8a5ffe77bf85706e7/jbpm-container-test/jbpm-in-container-test/jbpm-container-test-suite/src/test/java/org/jbpm/test/container/archive/EJBService.java#L261-L263

**Pull request merged**